typescript/sdk/src/deploy/router/AbacusRouterDeployer.ts

```diff
@@ -69,28 +69,31 @@ export abstract class AbacusRouterDeployer<
   async enrollRemoteRouters(
     contractsMap: ChainMap<Chain, Contracts>,
   ): Promise<void> {
-    this.logger(`Enrolling deployed routers with each other...`);
+    this.logger(
+      `Enrolling deployed routers with each other (if not already)...`,
+    );
     // Make all routers aware of each other.
+    const deployedChains = Object.keys(contractsMap);
     await promiseObjAll(
       objMap(contractsMap, async (local, contracts) => {
         const chainConnection = this.multiProvider.getChainConnection(local);
-        for (const remote of this.multiProvider.remoteChains(local)) {
-          const remoteRouterAddress = utils.addressToBytes32(
+        // only enroll chains which are deployed
+        const enrollChains = this.multiProvider
```

Comment:
Is this change backwards compat? Just to know for when we do the 0.5.0 release

Reply:
I would describe it as a bug fix
enrolling undeployed remote chains would cause the deploy to that chain to fail